### PR TITLE
Work around Drupal 10.1 CSS/JS aggregation issues.

### PR DESCRIPTION
**Pull request by kepol**
See d.o issue: https://www.drupal.org/project/quantcdn/issues/3390296

Note that Stu provided the basic functionality via Slack. I've tested it on 10.1 through UI and drush by seeding theme assets.

### CODE REVIEW

File: modules/quant_api/src/EventSubscriber/QuantApi.php [hunk 1]
```diff
@@ -177,14 +178,46 @@ public function onOutput(QuantEvent $event) {
...
         if (file_exists($fileOnDisk) && md5_file($fileOnDisk) == $item['existing_md5']) {
           continue;
         }
       }
 
-      // If the file exists we send it directly to quant otherwise we add it
-      // to the queue to generate assets on the next run.
+      // In Drupal 10.1, work around CSS/JS aggregation issues.
+      // Only process internal CSS/JS files.
+      $check_file = preg_match('/\.(css|js)$/', $file) && (strpos($item['original_path'], "http") !== 0);
```
Comment (kepol) [inline modules/quant_api/src/EventSubscriber/QuantApi.php]: I'm sure the `preg_match` could be updated to also check the `http` but didn't want to fuss with it... feel free to suggest a better regex.

File: modules/quant_api/src/EventSubscriber/QuantApi.php [hunk 2]
```diff
@@ -177,14 +178,40 @@ public function onOutput(QuantEvent $event) {
...
```
Comment (steveworley) [inline modules/quant_api/src/EventSubscriber/QuantApi.php]:
```suggestion
      $check_file = preg_match('/\.(css|js)$/', $file) && str_starts_with($item['original_path'], "http");
```
PHP8 introduces [str_starts_with](https://www.php.net/manual/en/function.str-starts-with.php) for readability we should look to use it. It's been polyfilled back to older versions as well so should be pretty safe to use.

https://core.trac.wordpress.org/ticket/58012

Comment (steveworley) [reply modules/quant_api/src/EventSubscriber/QuantApi.php]: It also introduced [str_ends_with](https://www.php.net/manual/en/function.str-ends-with.php) — this doesn't support multiple comparisons but would also be easier than the regex to maintain and using native language constructs may perform better.

```suggestion
      $check_file = (str_ends_with($file, 'css') || str_ends_with($file, 'js')) && str_starts_with($item['original_path'], 'http');
```

**Review by steveworley — CHANGES_REQUESTED** [review]
Minor change on the `strpos` and the regex — `str_starts_with` and `str_ends_with` make the code easier to grok.

Thanks!

File: modules/quant_api/src/EventSubscriber/QuantApi.php (hunk 2 above)
Comment (kepol) [reply modules/quant_api/src/EventSubscriber/QuantApi.php]: Ah, I saw that function but didn’t realize it was back filled so I was worried about using it. I’ll get this in and merged and create a new release with the 2 D10 fixes 👍

**Comment by kepol** [comment]
Adjust logic and tested again on 10.1 and it's working fine so I'll merge this.

**Comment by kepol** [comment]
Included in [1.5.4 release](https://www.drupal.org/project/quantcdn/releases/1.5.4).

**Pull request merged**